### PR TITLE
Introduce source_type flag

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Depends-On: https://github.com/thoth-station/common/pull/813
Depends-On: https://github.com/thoth-station/common/pull/815

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

- Introduce the source_type flag for Thoth integrations
- Remove is_s2i old flag

### CODE REVIEW

File: thoth/user_api/api_v1.py [hunk 1]
```diff
@@ -264,6 +266,12 @@ def post_advise_python(
     parameters = locals()
     parameters["application_stack"] = parameters["input"].pop("application_stack")
 
+    integrations = [i.value for i in ThothAdviserIntegrationEnum]
+
+    if source_type:
+        if source_type not in integrations:
+            raise NotKnownThothIntegration(f"This integration {source_type} is not provided in Thoth: {integrations}")
```
Comment (fridex) [inline thoth/user_api/api_v1.py:273]: You can put this exception directly to thoth-common to have it as close to error state as possible.

File: thoth/user_api/api_v1.py [hunk 2]
```diff
@@ -246,12 +247,12 @@ def get_provenance_python_status(analysis_id: str):
...
-    limit: int = None,
-    limit_latest_versions: int = None,
-    origin: str = None,
-    is_s2i: bool = None,
+    recommendation_type: typing.Optional[str] = None,
+    count: typing.Optional[int] = None,
+    limit: typing.Optional[int] = None,
+    limit_latest_versions: typing.Optional[int] = None,
+    origin: typing.Optional[str] = None,
+    source_type: Optional[ThothAdviserIntegrationEnum] = None,
```
Comment (KPostOffice) [inline thoth/user_api/api_v1.py:255]: typing.Optional here as well. Other than that looks good :+1:

File: thoth/user_api/api_v1.py [hunk 3]
```diff
@@ -246,12 +247,12 @@ def get_provenance_python_status(analysis_id: str):
...
-    limit: int = None,
-    limit_latest_versions: int = None,
-    origin: str = None,
-    is_s2i: bool = None,
+    recommendation_type: typing.Optional[str] = None,
+    count: typing.Optional[int] = None,
+    limit: typing.Optional[int] = None,
+    limit_latest_versions: typing.Optional[int] = None,
+    origin: typing.Optional[str] = None,
+    source_type: typing.Optional[ThothAdviserIntegrationEnum.value] = None,
```
Comment (KPostOffice) [inline thoth/user_api/api_v1.py:255]: Isn't `.value` an instance variable which returns the string representation of the enum?

Comment (pacospace) [reply thoth/user_api/api_v1.py]: updated :)

**Comment by goern** [comment]
regate

**Pull request merged**